lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -3803,6 +3807,17 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
     SmallVector<SILValue, 8> origFormalResults;
     collectAllFormalResultsInTypeOrder(original, origFormalResults);
     auto origResult = origFormalResults[task->getIndices().source];
+    // Emit warning if original result is not varied, because it will always have
+    // a zero derivative.
+    if (!activityInfo.isVaried(origResult, task->getIndices().source)) {
+      // Emit fixit if original result has a valid source location.
+      auto sourceLoc = origResult.getLoc().getSourceLoc();
```

Comment:
Does it have an `operator bool()`? Can you do `if (auto sourceLoc = ...` instead?

Reply:
It seems not.
`Value of type 'swift::SourceLoc' is not contextually convertible to 'bool'`